mlflow/java/client/src/main/java/org/mlflow/client/ApiClient.java

```diff
@@ -0,0 +1,140 @@
+  }
+
+  /** Returns a default client based on the MLFLOW_TRACKING_URI environment variable. */
+  public static ApiClient defaultClient() {
+    String defaultTrackingUri = System.getenv("MLFLOW_TRACKING_URI");
+    if (defaultTrackingUri == null) {
+      throw new IllegalStateException("Default client requires MLFLOW_TRACKING_URI is set." +
+        " Use fromTrackingUri() instead.");
+    }
+    return fromTrackingUri(defaultTrackingUri);
+  }
+
+  /** Instantiates a new client using the provided tracking uri. */
+  public static ApiClient fromTrackingUri(String trackingUri) {
+    URI uri = URI.create(trackingUri);
+    MlflowHostCredsProvider provider;
+     if ("http".equals(uri.getScheme()) || "https".equals(uri.getScheme())) {
+      provider = new BasicMlflowHostCreds(trackingUri);
+    } else if (uri.getScheme() == null || "file".equals(uri.getScheme())) {
+      throw new IllegalArgumentException("Java Client currently does not support" +
+        " local tracking URIs. Please point to a Tracking Server.");
+    } else {
+      throw new IllegalArgumentException("Invalid tracking server uri: " + trackingUri);
+    }
+    return new ApiClient(provider);
+  }
+
+  public GetExperiment.Response getExperiment(long experimentId) throws Exception {
+    URIBuilder builder = new URIBuilder("experiments/get")
+      .setParameter("experiment_id", "" + experimentId);
+    return toMapper.toGetExperimentResponse(httpCaller.get(builder.toString()));
+  }
+
+  public List<Experiment> listExperiments() throws Exception {
+    return toMapper.toListExperimentsResponse(httpCaller.get("experiments/list"))
+      .getExperimentsList();
+  }
+
+  public long createExperiment(String experimentName) throws Exception {
+    String ijson = fromMapper.makeCreateExperimentRequest(experimentName);
+    String ojson = httpCaller.post("experiments/create", ijson);
+    return toMapper.toCreateExperimentResponse(ojson).getExperimentId();
+  }
+
+  public Run getRun(String runUuid) throws Exception {
+    URIBuilder builder = new URIBuilder("runs/get").setParameter("run_uuid", runUuid);
+    return toMapper.toGetRunResponse(httpCaller.get(builder.toString())).getRun();
+  }
+
+  public RunInfo createRun(CreateRun request) throws Exception {
+    String ijson = fromMapper.toJson(request);
+    String ojson = post("runs/create", ijson);
+    return toMapper.toCreateRunResponse(ojson).getRun().getInfo();
+  }
+
+  public RunInfo createRun() throws Exception {
+    return createRun(CreateRun.newBuilder().build());
+  }
+
+  public void updateRun(String runUuid, RunStatus status, long endTime) throws Exception {
+    post("runs/update", fromMapper.makeUpdateRun(runUuid, status, endTime));
+  }
+
+  public void logParameter(String runUuid, String key, String value) throws Exception {
+    post("runs/log-parameter", fromMapper.makeLogParam(runUuid, key, value));
+  }
+
+  public void logMetric(String runUuid, String key, float value) throws Exception {
```

Comment:
Can we make the value a double? If the proto is using a float, we should change that to double too.

Reply:
Let's change to double in a follow-up, that probably has further implications. I'll create an issue.